extensions/mysql/lib/mysql.go

```diff
@@ -0,0 +1,439 @@
+				return cache.LastArchivedBinlog
+			}
+		}
+	}
+	if os.IsNotExist(err) {
+		tracelog.InfoLogger.Println("MySQL cache does not exist")
+	} else {
+		tracelog.ErrorLogger.Printf("%+v\n", err)
+	}
+	return ""
+}
+
+func setLastArchivedBinlog(binlogFileName string) {
+	var cache MySQLLogsCache
+	var cacheFilename string
+
+	usr, err := user.Current()
+	if err == nil {
+		cacheFilename = filepath.Join(usr.HomeDir, MysqlBinlogCacheFileName)
+		var file []byte
+		file, err = ioutil.ReadFile(cacheFilename)
+		// here we ignore whatever error can occur
+		if err == nil {
+			_ = json.Unmarshal(file, &cache)
+		}
+	}
+	if err != nil && !os.IsNotExist(err) {
+		tracelog.ErrorLogger.Printf("%+v\n", err)
+	}
+
+	cache.LastArchivedBinlog = binlogFileName
+
+	marshal, err := json.Marshal(&cache)
+	if err == nil && len(cacheFilename) > 0 {
+		_ = ioutil.WriteFile(cacheFilename, marshal, 0644)
+	}
+}
+
+func getMySQLSortedBinlogs(db *sql.DB) []string {
+	var result []string
+
+	currentBinlog := getMySQLCurrentBinlogFile(db)
+
+	rows, err := db.Query("show binary logs")
+	if err != nil {
+		tracelog.ErrorLogger.Fatalf("%+v\n", err)
+	}
+	for rows.Next() {
+		var logFinName string
+		var size uint32
+		err = rows.Scan(&logFinName, &size)
+		if err != nil {
+			tracelog.ErrorLogger.Fatalf("%+v\n", err)
+		}
+		if logFinName < currentBinlog {
+			result = append(result, logFinName)
+		}
+	}
+	sort.Strings(result)
+	return result
+}
+
+// TODO : unit tests
+func (backup *Backup) fetchStreamSentinel() (StreamSentinelDto, error) {
+	sentinelDto := StreamSentinelDto{}
+	backupReaderMaker := internal.NewStorageReaderMaker(backup.BaseBackupFolder,
+		backup.GetStopSentinelPath())
+	backupReader, err := backupReaderMaker.Reader()
+	if err != nil {
+		return sentinelDto, err
+	}
+	sentinelDtoData, err := ioutil.ReadAll(backupReader)
+	if err != nil {
+		return sentinelDto, errors.Wrap(err, "failed to fetch sentinel")
+	}
+
+	err = json.Unmarshal(sentinelDtoData, &sentinelDto)
+	return sentinelDto, errors.Wrap(err, "failed to unmarshal sentinel")
+}
+
+func getMySQLCurrentBinlogFile(db *sql.DB) (fileName string) {
+	rows, err := db.Query("SHOW MASTER STATUS")
+	if err != nil {
+		tracelog.ErrorLogger.Fatalf("%+v\n", err)
+	}
+	var logFileName string
+	var garbage interface{}
+	for rows.Next() {
+		err = rows.Scan(&logFileName, &garbage, &garbage, &garbage, &garbage)
+		if err != nil {
+			tracelog.ErrorLogger.Fatalf("%+v\n", err)
+		}
+		return logFileName
+	}
+	rows, err = db.Query("SHOW SLAVE STATUS")
+	if err != nil {
+		tracelog.ErrorLogger.Fatalf("%+v\n", err)
+	}
+	for rows.Next() {
+		err = rows.Scan(&logFileName, &garbage, &garbage, &garbage, &garbage)
+		if err != nil {
+			tracelog.ErrorLogger.Fatalf("%+v\n", err)
+		}
+		return logFileName
+	}
+	tracelog.ErrorLogger.Fatalf("Failed to obtain current binlog file")
+	return ""
+}
+
+func getMySQLConnection() (*sql.DB, error) {
+	datasourceName := internal.GetSettingValue("WALG_MYSQL_DATASOURCE_NAME")
```

Comment:
What about adding a default value parameter to GetSettingValue function?

Reply:
Done